ethdb/privateapi/ethbackend.go

```diff
@@ -196,9 +206,11 @@ func (s *EthBackendServer) EngineExecutePayloadV1(ctx context.Context, req *type
 	}
 
 	blockHash := gointerfaces.ConvertH256ToHash(req.BlockHash)
-
+	// Discard all previous prepared payloads if another block was proposed
+	s.nextPayloadId = 0
```

Comment:
Why is nextPayloadId set to zero every time EngineExecutePayloadV1 is called?

Reply:
if a new block is executed, it means that we can discard assembled block based on the previous state, nextPayloadId is set to 0 so that we do not risk overflows and what not, just seemed a good thing to do

Reply:
Resetting payload ID might be confusing to the Consensus Layer. It'll also complicate tracking and debugging of different payloads. If we manage to overflow 2^64, then something went very wrong. Beacon epoch is about 12 sec, and Engine API requests should come with roughly that frequency. Thus we have about 7*10^12 years as a safety margin :)

Reply:
that is fine to remove, it was more of personal preference, i do not think it gives problems to beacon node because if you turned off any go-ethereum then it would not remember which payloadId goes to which payload